Source/EasyNetQ/ConnectionString/ConnectionStringGrammar.cs

```diff
@@ -77,22 +81,19 @@ public static IEnumerable<UpdateConfiguration> ParseConnectionString(string conn
         Expression<Func<ConnectionConfiguration, T>> getter
     ) => CreateSetter<ConnectionConfiguration, T>(getter);
 
-    /// <summary>
-    /// Stolen from SO:
-    /// http://stackoverflow.com/questions/4596453/create-an-actiont-to-set-a-property-when-i-am-provided-with-the-linq-expres
-    /// </summary>
-    /// <typeparam name="TContaining"></typeparam>
-    /// <typeparam name="TProperty"></typeparam>
-    /// <param name="getter"></param>
-    /// <returns></returns>
     private static Action<TContaining, TProperty> CreateSetter<TContaining, TProperty>(Expression<Func<TContaining, TProperty>> getter)
     {
-        if (getter.Body is not MemberExpression memberEx) throw new ArgumentOutOfRangeException(nameof(getter), "Body is not a member-expression.");
-        if (memberEx.Member is not PropertyInfo propertyInfo) throw new ArgumentOutOfRangeException(nameof(getter), "Member is not a property.");
-        if (!propertyInfo.CanWrite) throw new ArgumentOutOfRangeException(nameof(getter), "Member is not a writeable property.");
-
-        var setMethodInfo = propertyInfo.GetSetMethod() ?? throw new ArgumentOutOfRangeException(nameof(getter), "No set method.");
-        return (Action<TContaining, TProperty>)setMethodInfo.CreateDelegate(typeof(Action<TContaining, TProperty>));
+        if (getter.Body is not MemberExpression memberEx)
+            throw new ArgumentOutOfRangeException(nameof(getter), "Body is not a member-expression");
+        if (memberEx.Member is not PropertyInfo propertyInfo)
+            throw new ArgumentOutOfRangeException(nameof(getter), "Member is not a property");
+        if (!propertyInfo.CanWrite)
+            throw new ArgumentOutOfRangeException(nameof(getter), "Member is not a writeable property");
+
+        var valueParameterExpr = Expression.Parameter(typeof(TProperty), "value");
```

Comment:
To support expressions with a single member access(`c => c.Name`) as well as with multiple ones(`c.Ssl.Enabled`).